messageflux/iodevices/sqs/sqs_input_device.py

```diff
@@ -66,82 +67,62 @@ def __init__(
 
         """
         super().__init__(device_manager, queue_name)
-        self._included_message_attributes = (
-            included_message_attributes
-            if included_message_attributes is not None
-            else ["All"]
-        )
-        self._max_messages_per_request = 1
+
+        if included_message_attributes is None:
+            included_message_attributes = ["All"]
+
+        self._included_message_attributes = included_message_attributes
+        self._max_messages_per_request = 1  # TODO: get this in manager
         self._queue = self.manager.get_queue(queue_name)
 
     def _read_message(
-        self,
-        cancellation_token: threading.Event,
-        timeout: Optional[float] = None,
-        with_transaction: bool = True,
+            self,
+            cancellation_token: threading.Event,
+            timeout: Optional[float] = None,
+            with_transaction: bool = True,
     ) -> Optional["ReadResult"]:
         """
         reads a stream from InputDevice (tries getting a message. if it fails, reconnects and tries again once)
 
         :param timeout: the timeout in seconds to block. negative number means no blocking
         :return: a tuple of stream and metadata from InputDevice, or (None, None) if no message is available
         """
-
-        sqs_messages = self._queue.receive_messages(
-            MessageAttributeNames=["All"],
-            MaxNumberOfMessages=self._max_messages_per_request,
-        )
+        if timeout is None:
+            sqs_messages = self._queue.receive_messages(
+                MessageAttributeNames=self._included_message_attributes,
+                MaxNumberOfMessages=self._max_messages_per_request,
+            )  # TODO: what's the visibility timeout? should we extend it?
```

Comment:
The visibility timeout usually configured in the queue created and it is default to the queue.
Why do we want to extend is?

Reply:
If, as the developer of the pipeline, the max time you give to process message is X, then you should configure X as the visibility timeout for the queue. 
IMO, this decision is out of the scope of the sqs input device implementation.